futures-util/src/stream/futures_unordered/ready_to_run_queue.rs

```diff
@@ -85,25 +85,34 @@ impl<Fut> ReadyToRunQueue<Fut> {
     pub(super) fn stub(&self) -> *const Task<Fut> {
         &*self.stub
     }
+
+    // Clear the queue of tasks.
+    //
+    // # Safety
+    //
+    // Note that each task has a strong reference count associated with it
+    // which is owned by the ready to run queue. All tasks MUST have had
+    // their futures dropped already by `FuturesUnordered::clear`. This
+    // method just pulls out tasks and drops their refcounts.
+    pub(crate) unsafe fn clear(&self) {
+        loop {
+            match self.dequeue() {
```

Comment:
Then how does `FuturesUnorder::clear` call `self.ready_to_run_queue.clear()` through the `Arc`?

Reply:
I think the best we can do is add a safety requirement of unique access. The current `self` still has weak references to itself through it's tasks, so `Arc::get_mut` will not work.

Reply:
Oh, sorry. I forgot it is wrapped in `Arc`.

> I missed that requirement.

I think that safety requirement is guaranteed by `&mut self` of `FuturesUnorder::clear`, but it would be nice to have code comments on that.

Reply:
> I think the best we can do is add a safety requirement of unique access.

Agreed.